core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/DoubleTerms.java

```diff
@@ -99,6 +100,16 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.endObject();
             return builder;
         }
+
+        @Override
+        public boolean equals(Object obj) {
+            return super.equals(obj) && Objects.equals(term, ((Bucket) obj).term);
```

Comment:
This probably needs the traditional class comparison to make sure the cast doesn't blow up.